File: programmers/java/lev0/이차원 배열 대각선 순회하기.java
import java.util.stream.IntStream;

class Solution {
    public int solution(int[][] board, int k) {
        return IntStream.range(0, board.length).filter(i -> i <= k).map(i -> IntStream.range(0, board[i].length).filter(j -> j <= k - i).map(j -> board[i][j]).sum()).sum();

    }
}

File: programmers/js/lev0/이차원 배열 대각선 순회하기.js
function solution(board, k) {
  return board
    .flatMap((row, i) => row.filter((value, j) => i + j <= k))
    .reduce((acc, cur) => acc + cur, 0);

  //     return board.reduce(
  //   (total, row, i) =>
  //     total + row.reduce((prev, num, j) => (i + j <= k ? prev + num : prev), 0),
  //   0
  // );
}
